src/base/geom/affinetransform.cpp

```diff
@@ -7,108 +7,92 @@
 
 using namespace Geom;
 
-AffineTransform::AffineTransform()
-	: offset(0,0), scale(1.0), rotate(0.0)
+AffineTransform::AffineTransform() :
+	m{
+		{ 1,0,0 },
+		{ 0,1,0 }
+	}
 {}
 
-AffineTransform::AffineTransform(Point offset, double scale, double rotate) :
-    offset(offset),
-    scale(scale),
-    rotate(rotate)
+AffineTransform::AffineTransform(
+	double m00, double m01, double m02,
+	double m10, double m11, double m12) :
+	m {
+		{ m00, m01, m02 },
+		{ m10, m11, m12 }
+	}
 {}
 
-AffineTransform::AffineTransform(Rect from, Rect to)
-{
-	rotate = 0.0;
-	scale = std::max(to.size.x / from.size.x, to.size.y / from.size.y);
-	offset = to.pos - from.pos * scale;
-	to.pos = from.pos * scale + offset;
-	to.size = from.size * scale;
-}
+AffineTransform::AffineTransform(Point offset, double scale, double angle) :
```

Comment:
create static builder function as this constructor replacement.

Reply:
why?

Reply:
Not required. I just thought that the parameters are not simply as first seems, and it is less expressive than a `AffineTransformation::from_units(point, double, double)` function next to an `AffineTransformation::AffineTransformation(double, double, double...)` constructor